APE6.rst

```diff
@@ -0,0 +1,499 @@
+
+Supporting more complex metadata structures would be possible within the IPAC
+standard using keyword values, but it would be cumbersome and require inventing
+a custom serialization method to work within that confine.
+
+VOTable
+^^^^^^^^
+
+`VOTable <http://www.ivoa.net/documents/latest/VOT.html>`_ is by design a
+fully-flexible data format that can handle all of the needs for text
+serialization of complex data structures, including tablular data sets.  The
+issue in this context is in simplicity and data interchange with the broader
+community.  In essence if someone wants to read or write a VOTable then they
+must use one of a small number of implementations of this protocol.  It is not
+possible for someone to directly read such a table into Excel.  Writing an
+implementation of VOTable in a new language (e.g. R, Julia, or Perl) to read/write
+VOTable is a major undertaking.
+
+A smaller issue is speed, since the VOTable format requires relatively complex
+parsing and validation for text serialization.
+
+Others
+^^^^^^^^
+
+We are not aware of other widely-used standards in the astronomical
+community for text representation of tabular data.
+
+Tabular Data Package
+^^^^^^^^^^^^^^^^^^^^^
+
+Though not widely used in the astronomical community, since initially drafting
+this APE we have become aware of a very similar standard known as the
+`Tabular Data Package
+<http://dataprotocols.org/tabular-data-package/>`_.  This provides a
+fully-formed protocol for publishing and sharing tabular-style data
+which is conceptually very similar to the proposed ECSV format, with
+the exception of using two files, one pure JSON for the header and one
+pure CSV for the data.  The JSON header follows a schema defined by
+the `JSON Table Schema
+<http://dataprotocols.org/json-table-schema/>`_.
+
+
+After evaluation and a brief `discussion with that community
+<https://lists.okfn.org/pipermail/data-protocols/2014-April/000093.html>`_, we
+find that the Tabular Data Package and associated standards would require a
+fair degree of modification to fully suit our needs.  More crucially, the
+decision to rely on YAML instead of JSON for serialization precludes direct use
+of the TDP.
+
+
+Detailed description
+---------------------
+
+The proposed Enhanced CSV (ECSV) format has the following
+overall structure:
+
+- A header section which consists of lines that start with the ``#`` character
+  and provide the table definition and data format via a YAML-encoded data structure.
+- An initial line in the header section which identifies the file as ECSV and
+  provides a version number.
+- A CSV-formatted data section in which the first line contains the column names
+  and subsequent lines contains the data values.
+
+Version 0.9 of the ECSV format specification and the reference Python
+implementation assumes ASCII-encoded header and data sections.  Support
+for unicode (in particular UTF-8) may be added in subsequent versions.
```

Comment:
Speaking of encodings, would we allow separate encodings for the data and header?  I think this could be useful.  Or should that discussion be tabled?